store/orm/orm.go

```diff
@@ -66,10 +67,37 @@ func NewORM(uri string, timeout time.Duration) (*ORM, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to init DB: %+v", err)
 	}
-	return &ORM{
+
+	orm := &ORM{
 		DB:              db,
 		lockingStrategy: lockingStrategy,
-	}, nil
+	}
+
+	if linkstatsURL != nil {
+		db.Callback().Create().After("gorm:update").Register("sync:run_after_create", createSyncEvents)
```

Comment:
Personally, I would prefer to keep the ORM focused on the single responsibility of wrapping the db persistence layer. Could we pull this up just one layer to the `store` and have a separate type (`StatsPusher` or a new `SyncEventManager`?) set up the callback `createSyncEvents` via the public `DB`? This way, if the logic in `createSyncEvents` grows, we don't intrude on `orm.go`.

That being said, perhaps you see sync events as part of ORM.

Reply:
I've put the callback into another file, to separate it.

Reply:
Hey there, don't fully understand where the complexity comes in. The `scope.DB` belongs to `gorm` which can easily get passed in the `store` package or any other package without a circular dependency. Couldn't you just put this (and `createSyncEvents`) in the `StatsPusher (store/stats_pusher.go)` constructor with little hassle?

```golang
// NewStatsPusher returns a new event queuer
func NewStatsPusher(orm *orm.ORM, url *url.URL) *StatsPusher {
	var wsClient WebSocketClient
	wsClient = noopWebSocketClient{}
	if url != nil {
		wsClient = NewWebSocketClient(url)
                orm.DB.Callback().Create().After("gorm:update").Register("sync:run_after_create", createSyncEvents)
	}
```

Perhaps it's a reluctance to place `gorm` outside of `orm`. Will sync with you offline.